### PR TITLE
Bonita versioning

### PR DESCRIPTION
**Pull request by florentsarat**
Add a page on the Bonita version logic

### CODE REVIEW

File: md/bonitaversioning.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Bonita Versioning
+
```
Comment (rbioteau) [inline md/bonitaversioning.md]: A short description should be made to improve SEO

File: md/bonitaversioning.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+## Product Versions
+
+Bonita follows the usual semantic versioning X.Y.z:
+* X is the major version number. The major version changes when non-backward compatible changes are implemented
+* Y is the minor version number. The minor version changes when backward compatible features are added. A minor version is released every 6 months to improve continuously Bonita
+* z being the maintenance version number. The maintenance versions contain fixes. Maintenance versions are released every month between two minor versions for the current minor version.
```
Comment (rbioteau) [inline md/bonitaversioning.md]: and are cross-compatible by default for the same minor version ?

File: md/bonitaversioning.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+## Artifact version
+
+In addition to the product version, each artifact has a model version. This model version is used to determine if the corresponding artifact is compatible with the development suite (Studio & UI Designer).
```
Comment (rbioteau) [inline md/bonitaversioning.md]:
```suggestion
In addition to the product version, most artifacts have a model version. This model version is used to determine if the corresponding artifact is compatible with the development suite (Studio & UI Designer).
```

File: md/bonitaversioning.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+If the artifacts are not compatible then either the artifact is migrated (when the model version lower than the development suite version) or you will have to upgrade your development suite.
```
Comment (rbioteau) [inline md/bonitaversioning.md]:
```suggestion
If the artifacts are not compatible then either the artifact is migrated (when the model version is lower than the development suite version) or you will have to upgrade your development suite to a compatible version.
```

File: md/bonitaversioning.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+The following artifacts have model versions:
+* Diagrams
+* Organization
+* Business Data Model
```
Comment (rbioteau) [inline md/bonitaversioning.md]: BDM Access control

File: md/bonitaversioning.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+* Applications
```
Comment (rbioteau) [inline md/bonitaversioning.md]:
```suggestion
* Applications descriptors
```

File: md/bonitaversioning.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+* Pages
+* Forms
+* Layouts
+* Customer Widgets
```
Comment (rbioteau) [inline md/bonitaversioning.md]: Fragments

File: md/bonitaversioning.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+These model versions should not change between maintenance versions of a given minor version (e.g. between a 7.12.5 and a 7.12.3). This means that no migration is needed in production or the studio when changing the maintenance version.
+
+::: info
+**Note:** Changing the minor version does not imply that the model version all artifacts will change
```
Comment (rbioteau) [inline md/bonitaversioning.md]:
```suggestion
**Note:** Changing the minor version does not imply that the model version of all artifacts will change
```

File: md/taxonomy.md [hunk 9]
```diff
@@ -160,6 +160,7 @@
     * [Ubuntu + OpenJDK + Tomcat + PostgreSQL](ubuntu-openjdk-tomcat-postgresql.md)
     * [Install Tomcat with Bonita as a service in Windows](bonita-as-windows-service.md)
   * [Back up Bonita Platform](back-up-bonita-bpm-platform.md)
+  * [Bonita Versions](bonitaversioning.md)
```
Comment (rbioteau) [inline md/taxonomy.md]:
```suggestion
  * [Bonita Versions](product-versioning.md)
```

File: md/bonitaversioning.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+:::
+
+## Version checks in the development suite
+
+### Studio
+The Studio will check the model version of every artifact of a given project when cloning or importing it. The artifact model version will also be checked when importing artifacts unitarily. 
+
+The model versions of all artifacts are also checked when deploying a project. Finally, the model version is checked when an artifact is opened.
+
+To check the compatibility of your artifacts manually, you can use the "Validate" action in the project's contextual menu.
```
Comment (rbioteau) [inline md/bonitaversioning.md]:
```suggestion
To check the compatibility of your artifacts manually, you can use the "Validate" action in the project's contextual menu or via the File menu.
```

File: md/bonitaversioning.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+You will not able to work on incompatible artifacts or deploy them. You will be able if you desire to migrate older artifacts.
+
```
Comment (rbioteau) [inline md/bonitaversioning.md]: add a warning on the impact of migrating artifacts for other users of the same project

**Review by rbioteau — CHANGES_REQUESTED** [review]
Use `product-versioning.md` filename for a better url path ?

**Review by rbioteau — APPROVED** [review]

File: md/product-versioning.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
+# Bonita Versioning
+Discover how Bonita manages its product and artifacts versions and how it can impact your projects
+
+## Product Versions
+
+Bonita follows the usual semantic versioning X.Y.z:
+* X is the major version number. It changes when non-backward compatible changes are implemented.
```
Comment (jeromecambon) [inline md/product-versioning.md:7]: Is it always true? Can't we release a 8.0 without compatibility break?

File: md/product-versioning.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+* Y is the minor version number. It version changes when backward compatible features are added. A minor version is released every six months to improve continuously Bonita.
+* z is the maintenance version number. The maintenance versions contain fixes. Maintenance versions are released every month between two minor versions for the current minor version. These versions are cross-compatible by default for the same minor version.
```
Comment (jeromecambon) [inline md/product-versioning.md]: Why do we have a lowercase z? Seems like a typo.
I would add an example to make the things very clear: not sure "cross-compatible" is obvious for everybody.

File: md/product-versioning.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+## Artifact version
+
+In addition to the product version, most artifacts have a model version. The development suite (Studio and UID) uses this model version to determine if the corresponding artifact is compatible.
```
Comment (jeromecambon) [inline md/product-versioning.md]: In the external docs, we always say "UI Designer", and never use "UID"
"... to determine if the artifacts are compatible."

File: md/product-versioning.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+If the artifacts are incompatible, then either the development suite migrates the artifact (when the model version is lower than the development suite version), or you will have to upgrade your development suite to a compatible version.
```
Comment (jeromecambon) [inline md/product-versioning.md]: "... the development suite migrates the artifact**S** ..."

Comment (jeromecambon) [reply md/product-versioning.md]: Please add examples to make it very clear.

File: md/product-versioning.md [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+* Pages
+* Forms
+* Fragments
+* Layouts
+* Customer Widgets
+
+These model versions should not change between maintenance versions of a given minor version (e.g. between a 7.12.5 and a 7.12.3). This means that no migration is needed in production or the Studio when changing the maintenance version.
+
+::: info
+**Note:** Changing the minor version does not imply that the model version of all artifacts will change
```
Comment (jeromecambon) [inline md/product-versioning.md]: "Changing the minor version does not imply that the model version will change. This means that no migration will be needed in this case."

File: md/product-versioning.md [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+### Studio
+The Studio will check the model version of every artifact of a given project when cloning or importing it. The artifact model version will also be checked when importing artifacts individually. 
+
+The model versions of all artifacts are also checked when deploying a project. Finally, the model version is checked when an artifact is opened.
+
+To check the compatibility of your artifacts manually, you can use the "Validate" action in the project's context menu or via the File menu.
+
+You will not be able to work with incompatible artifacts or deploy them. You will be able if you desire to migrate to older artifacts.
+
+### UID
```
Comment (jeromecambon) [inline md/product-versioning.md]: UI Designer

File: md/product-versioning.md [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+
+The UID will check the model version of all artifacts when it is launched and when a given artifact is opened. You will not be able to work with incompatible artifacts. You will be able if you desire to migrate to older artifacts.
```
Comment (jeromecambon) [inline md/product-versioning.md]: idem

Comment (jeromecambon) [reply md/product-versioning.md]: Remind what is an incompatible artifact.
" You will be able if you desire to migrate to older artifacts." => I guess you mean "FROM older artifacts" ?

**Review by jeromecambon — CHANGES_REQUESTED** [review]
Good doc ! Few comments here and there.

File: md/product-versioning.md (hunk 12 above)
Comment (florentsarat) [reply md/product-versioning.md:7]: Anything is possible. I rather be restrictive here.

**Review by jeromecambon — APPROVED** [review]

**Pull request merged**